src/inmanta/server/services/orchestrationservice.py

```diff
@@ -517,6 +624,38 @@ async def _put_version(
         *,
         connection: asyncpg.connection.Connection,
     ) -> None:
+        """
+        :param rid_to_resource: This parameter should contain all the resources when a full compile is done.
+                                When a partial compile is done, it should contain all the resources that belong to the
+                                updated resource sets or the shared resource sets.
+        :param unknowns: This parameter should contain all the unknowns for all the resources in the new version of the model.
+                         Also the unknowns for resources that are not present in rid_to_resource.
+        :param partial_base_version: When a partial compile is done, this parameter contains the version of the
+                                     configurationmodel this partial compile was based on. Otherwise this parameter should be
+                                     None.
+        :param removed_resource_sets: When a partial compile is done, this parameter should indicate the names of the resource
+                                      sets that are removed by the partial compile. When no resource sets are removed by
+                                      a partial compile or when a full compile is done, this parameter can be set to None.
+
+        Pre-conditions:
```

Comment:
Perhaps add a section for validations performed by this method as well?
- In case of a full export, checks whether this version has any requires-provides across resource sets and sets the `is_suitable_for_partial_compiles` field appropriately, indicating whether this version is eligible to be used as a base version for a future partial compile.
- In case of a partial export, verifies that no resources moved resource sets.